src/filebrowser/dialogs.ts

```diff
@@ -291,6 +291,7 @@ class CreateFromHandler extends Widget {
           return widget;
         });
       }
+      this._model.deleteFile(this._orig);
```

Comment:
Good catch @dhuebner.  We added signals to the `ContentsManager` in `@jupyterlab/services` that allow the `FileBrowserModel` to refresh automatically when one of the files in its directory changes.